File: lotteryticket/src/main/java/com/wuhai/lotteryticket/ui/base/NewLoadingBaseActivity.java
package com.wuhai.lotteryticket.ui.base;

import android.os.Bundle;

import com.wuhai.lotteryticket.widget.dialog.LoadingDialog;


/**
 * new loading BaseActivity
 */
public abstract class NewLoadingBaseActivity extends BaseActionBarActivity {

    public LoadingDialog mLoading;//new loading dialog

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mLoading = new LoadingDialog(this);
    }

    /**
     * showNewLoading
     * ※跟 基类里的showLoading() 区别，用基类showLoading()，成功后必须调showSuccessLayout()
     */
    public void showNewLoading(){
        if (mLoading != null && !mLoading.isShowing()) {
            try {
                mLoading.show();
            } catch (Exception e) {
            }

        }
    }

    /**
     * dimssNewLoading
     */
    public void dimssNewLoading() {
        if (mLoading != null && mLoading.isShowing()) {
            try {
                mLoading.dismiss();
            } catch (Exception e) {

            }

        }
    }

    @Override
    public void onDestroy() {
        dimssNewLoading();
        mLoading = null;
        super.onDestroy();
    }

    /**
     *
     * @param msg
     */
//    public void showErrorMsg(String msg) {
//        DSDialog dialog = new DSDialog.Builder(this).setType(DSDialog.TYPE_BOTTOM_ERROR).setTitle("错误提示")
//                .setMessage(msg).setPositiveButtonText("知道了").setButtonClickListener(new DSDialog.ActionClickListener() {
//                    @Override
//                    public void onClick(DSDialog dialog, int which, Object data) {
//                        dialog.dismiss();
//                    }
//                }).setIsBackCancelable(false).create();
//
//        if (!this.isFinishing()) {
//            dialog.show();
//        }
//    }
}

File: lotteryticketkotlin/src/main/java/com/wuhai/demo/lotteryticketkotlin/ui/base/NewLoadingBaseActivity.kt
package com.wuhai.demo.lotteryticketkotlin.ui.base

import android.os.Bundle
import com.wuhai.demo.lotteryticketkotlin.widget.dialog.LoadingDialog

abstract class NewLoadingBaseActivity: BaseActionBarActivity() {
    var mLoading //new loading dialog
            : LoadingDialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mLoading = LoadingDialog(this)
    }

    /**
     * showNewLoading
     * ※跟 基类里的showLoading() 区别，用基类showLoading()，成功后必须调showSuccessLayout()
     */
    open fun showNewLoading() {
        //TODO 这样并不是java 对应的&&的逻辑呢
//        if ((mLoading != null).and(mLoading!!.isShowing)) {
//            try {
//                mLoading?.show()
//            } catch (e: Exception) {
//            }
//        }

        if(mLoading != null){
            if(mLoading!!.isShowing){
                mLoading?.show()
            }
        }
    }

    /**
     * dimssNewLoading
     */
    open fun dimssNewLoading() {
        if(mLoading != null){
            if(mLoading!!.isShowing){
                mLoading?.dismiss()
            }
        }
    }

    override fun onDestroy() {
        dimssNewLoading()
        mLoading = null
        super.onDestroy()
    }
}